### PR TITLE
Feat: TTS provider plugin check cache before synthesizing

### PR DESCRIPTION
**Pull request by xquanluu**

### CODE REVIEW

**Comment by xquanluu** [comment]
Hi @psanders  I created here a small draft version of #333, please help to check and give comment. thanks

File: mods/tts/src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { SynthResult, TTSConfig, TTSPlugin } from "./types";
```
Comment (psanders) [inline mods/tts/src/plugin.ts:20]: Could you add the copyright notice to this file? Feel free to add a note including your contribution.

**Review by psanders — CHANGES_REQUESTED** [review]
All changes seem great to me.

File: mods/marytts/src/types.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { TTSConfig } from "@fonoster/tts";
+
 /*
```
Comment (psanders) [inline mods/marytts/src/types.ts]: Would you move the copyright notice to the very top?

**Comment by psanders** [comment]
@efraa since this touches a full feature I would also like to have your review. Everything looks great to me, so I want to focus in making sure it doesn't introduce any breaking changes.

**Comment by xquanluu** [comment]
Thanks @psanders  I fixed your comments.

**Review by efraa — APPROVED** [review]

**Comment by efraa** [comment]
LGTM!

**Review by psanders — APPROVED** [review]

**Comment by psanders** [comment]
@xquanluu thanks again for this amazing collaboration 🙌

**Pull request merged**